### ISSUE TITLE
Improve the Celery setup on the development environment.

### ISSUE THREAD

**Issue by bugzilla-to-github** [mozilla/pontoon#2208]
*This issue was created automatically by a [script](https://github.com/mathjazz/bugzilla2github/).*
## [Bug 1612580](https://bugzilla.mozilla.org/show_bug.cgi?id=1612580)
Bug Reporter: @jotes
CC:   @adngdb, @Pike

The configuration of Pontoon's development environment doesn't include all of the components that are required by the Celery instance on Prod.
It makes issues like e.g. #1477570 harder to diagnose because e.g. you don't see if Celery is fetching tasks correctly.

I think it would be valuable addition to our repository. 

Pros: 
* development environment is closer to the state on Prod
Cons:
* Containers for RabbitMQ and Celery will consume additional resources

Additionaly, I would still run tests with `CELERY_ALWAYS_EAGER=True` because, without that, it will make testing a little bit harder and would require additional amount of work.

I've implemented a PoC of this setup here: https://github.com/mozilla/pontoon/compare/master...jotes:heroku-test-sync-ampq

Basically, I added all required components (RabbitMQ and a Celery worker) in a separate Dockerfile.

:Pike, :adrian What do you think about this?
